modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesChannel.java

```diff
@@ -185,7 +234,9 @@ public void recordRemovalFromBuffer(int size) {
 
         int count = messagesOnBuffer.addAndGet(-size);
 
-        if (flowControlEnabled && !globalFlowControlEnabled && count <= flowControlLowLimit) {
+        if (flowControlEnabled && (!globalBufferBasedFlowControlEnabled) 
+                               && (count <= flowControlLowLimit) 
+                               && (! globalErrorBasedFlowControlEnabled)) {
```

Comment:
This condition seems to be used everywhere. Let's make a private method and reuse? Then it is handy if we have to change the logic someday..